src/parameters_loader.rs

```diff
@@ -19,8 +19,66 @@ pub struct ParametersValues {
 }
 define_global_property!(Parameters, ParametersValues);
 
+/// Check whether parameters provided in input json
+/// are valid using our knowledge of allowable parameter
+/// values. This prevents us from, say, loading the synth
+/// pop only to realize that we accidentally set the GI to
+/// be a negative number, which would cause a panic much
+/// later in the code (when transmission happens).
+fn validate(parameters: &ParametersValues) -> Result<(), IxaError> {
+    if parameters.r_0 < 0.0 {
+        return Err(IxaError::IxaError(
+            "r_0 must be a non-negative number".to_string(),
+        ));
+    }
+    // need to think about a better way of validation
+    // in the long term that makes sense for an arbitrary GI
+    if parameters.generation_interval <= 0.0 {
+        return Err(IxaError::IxaError(
+            "r_0 must be a non-negative number".to_string(),
```

Comment:
It sounds like we don't know yet how to specify generation interval. Should we remove this for now and only leave `r_0`? Or at least change the error so that it doesn't say `r_0`

Reply:
Oops good catch, thanks! I like having this code there as a template for now and then we can change it as we decide on an input schema for the GI if you are ok with that?